pkg/cli/image/imagesource/dryrun.go

```diff
@@ -5,8 +5,8 @@ import (
 	"net/http"
 
 	"github.com/distribution/distribution/v3"
-	"github.com/distribution/distribution/v3/reference"
-	registryclient "github.com/distribution/distribution/v3/registry/client"
+	"github.com/distribution/reference"
+	registryclient "github.com/openshift/library-go/pkg/registry/client"
```

Comment:
Our main motivation is to bump the distribution in this PR?. Why we put ourselves under the pressure of moving to library-go as well?. I'd prefer first we have a plain distribution bump in one PR and only after we merge it, we can switch to library-go in a followup PR?.

Reply:
oc-mirror needs to upgrade distribution/distribution, but oc-mirror has indirect dependency of library-go and oc. So in order to upgrade distribution/distribution in oc-mirror we need to upgrade library-go and oc as well. Since oc depends on libray-go that's why I opened the PR to library-go first and added the removed packages of distribution/distribution there (since it is a common package for several components).

If we try to bump distribution/distribution in oc first, it will break saying that the new distribution/distribution versions does not have the code required by library-go.